frontend/src/employee-mobile-frontend/components/attendances/child-info/ImageEditor.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useRef, useState } from 'react'
+import ReactCrop from 'react-image-crop'
+import { defaultMargins, Gap } from '../../../../lib-components/white-space'
+import { FixedSpaceRow } from '../../../../lib-components/layout/flex-helpers'
+import Button from '../../../../lib-components/atoms/buttons/Button'
+import { uploadChildImage } from '../../../api/childImages'
+
+import 'react-image-crop/dist/ReactCrop.css'
+import styled from 'styled-components'
+import { useHistory } from 'react-router-dom'
+import { useTranslation } from '../../../state/i18n'
+
+const defaultCrop: ReactCrop.Crop = {
+  unit: '%',
+  width: 80,
+  x: 10,
+  y: 10,
+  aspect: 1
+}
+
+interface Props {
+  childId: string
+  image: string
+  onReturn: () => void
+}
+export default React.memo(function ImageEditor({
+  childId,
+  image,
+  onReturn
+}: Props) {
+  const { i18n } = useTranslation()
+  const [crop, setCrop] = useState<ReactCrop.Crop>(defaultCrop)
+  const [completedCrop, setCompletedCrop] = useState<ReactCrop.Crop | null>(
+    null
+  )
+  const [imageElem, setImageElem] = useState<HTMLImageElement | null>(null)
+  const previewCanvasRef = useRef<HTMLCanvasElement>(null)
+  const [submitting, setSubmitting] = useState(false)
+  const h = useHistory()
+
+  useEffect(() => {
+    const popStateHandler = (e: PopStateEvent) => {
+      e.preventDefault()
+      h.goForward()
+      onReturn()
+    }
+
+    window.addEventListener('popstate', popStateHandler)
+
+    return () => {
+      window.removeEventListener('popstate', popStateHandler)
+    }
+  }, [onReturn, h])
+
```

Comment:
this feels like a hack and causes warning in console. is there a better way to prevent back navigation and call `onReturn` instead?